### PR TITLE
[Tests-Only] adjust pact interactions for fileTrash and fileVersion tests

### PR DESCRIPTION
**Pull request by dpakach**
Related - https://github.com/owncloud/owncloud-sdk/issues/780

Fix interaction for pact tests
- [x] - FileTrashTest
- [x] - fileVersionTest

### CODE REVIEW

File: tests/fileVersionTest.js [hunk 1]
```diff
@@ -69,32 +61,43 @@ describe('Main: Currently testing file versions management,', function () {
...
+        .given('the user is recreated', {
+          username: testUser,
+          password: testUserPassword
+        })
+        .given('set random versionId')
+        .uponReceiving(`as '${testUser}', a PROPFIND request to get file versions of non existent file`)
+        .withRequest({
+          method: 'PROPFIND',
+          path: MatchersV3.regex(
+            '.*\\/remote\\.php\\/dav\\/meta\\/42\\/v', /* eslint-disable-line no-template-curly-in-string */
```
Comment (saw-jan) [inline tests/fileVersionTest.js:79]: `/* eslint-disable-line no-template-curly-in-string */` did you forget to remove?

File: yarn.lock [hunk 2]
```diff
@@ -1045,20 +1050,21 @@
...
-"@pact-foundation/pact-node@^10.10.1":
-  version "10.11.0"
-  resolved "https://registry.yarnpkg.com/@pact-foundation/pact-node/-/pact-node-10.11.0.tgz#f5e24da6ecd1d5b29c41e09958ce5eb9ac71eb76"
-  integrity sha512-dSGFSUjNCqsqWV23GY7ECuuhCazSDh902M1ZWDVl1g8HLED9Q4j29PsU0pnzZfCErgo21VYyb9yIXobBd3Hk3g==
+"@pact-foundation/pact-node@^10.11.4":
+  version "10.11.11"
+  resolved "https://registry.yarnpkg.com/@pact-foundation/pact-node/-/pact-node-10.11.11.tgz#c7b9b07f9661d6f5350ef9b364073eb767a4876a"
+  integrity sha512-G1I5hFX73BZZkiaXEz6u9SjgEEgFPUO6ZaTq6RqZuvZ+E6ZizantKpTOouoobWwB+dXnSiqU/3VB/jbNqZO5Cw==
   dependencies:
+    "@types/pino" "^6.3.5"
```
Comment (saw-jan) [inline yarn.lock:1058]: Are changes in `yarn.lock` expected to be committed?

File: tests/webdavHelper.js [hunk 3]
```diff
@@ -129,12 +132,95 @@ const getSignKey = function (username, password) {
...
 }
 
+/**
+ *
+ * @param {string} path
+ * @param {string} userId
+ * @param {array} properties
+ * @param {number} folderDepth
+ */
+const propfind = function (path, userId, password, properties, type = 'files', folderDepth = '1') {
```
Comment (saw-jan) [inline tests/webdavHelper.js:142]: why `password` and `type` not in doc

File: tests/providerTest.js [hunk 4]
```diff
@@ -130,14 +138,39 @@ describe('provider testing', () => {
           assertFoldersCreatedSuccessfully(results, dirname)
         }
         const result = createFile(
-          parameters.username, parameters.password, parameters.fileName, config.testContent
+          parameters.username, parameters.password, parameters.fileName, content
         )
         chai.assert.isBelow(
           result.status, 300, `creating file '${parameters.fileName}' failed`
         )
+        // delay(5000)
```
Comment (saw-jan) [inline tests/providerTest.js:146]: ??

File: tests/webdavHelper.js [hunk 5]
```diff
@@ -129,12 +132,95 @@ const getSignKey = function (username, password) {
...
+ *  "href":
+ *  "originalFilename":
+ *  "originalLocation":
+ *  "deleteTimestamp":
+ *  "lastModified":
+ * },...]
+ *
+ * @param {string} user
+ */
+const getTrashBinElements = function (user, password, depth = 2) {
```
Comment (saw-jan) [inline tests/webdavHelper.js:179]: also include `password` and `depth` in doc

File: tests/fileVersionTest.js [hunk 6]
```diff
@@ -158,35 +232,36 @@ describe('Main: Currently testing file versions management,', function () {
...
-          expect(versions.length).toEqual(2)
-          expect(versions[0].getSize()).toEqual(2)
-          expect(versions[1].getSize()).toEqual(1)
-        })
-        await oc.fileVersions.getFileVersionContents(fileInfo.id, fileInfo.versions[0].versionId).then(content => {
-          expect(content).toBe(fileInfo.versions[0].content)
-        })
-        return oc.fileVersions.getFileVersionContents(fileInfo.id, fileInfo.versions[1].versionId).then(content => {
-          expect(content).toBe(fileInfo.versions[1].content)
+        return provider.executeTest(async () => {
```
Comment (saw-jan) [inline tests/fileVersionTest.js:244]:
```suggestion
        await provider.executeTest(async () => {
```
else the test will only run once

**Comment by saw-jan** [comment]
LGTM :+1:

**Review by individual-it — CHANGES_REQUESTED** [review]
please use `createProvider(false, true)` for provider tests that are passing now on oc10

**Comment by dpakach** [comment]
@individual-it already done that in https://github.com/owncloud/owncloud-sdk/pull/798/files#diff-d4d0ad5a7c6c24e4ff2dd6a7da214b2219a1977b1ef376f315a95c5874d99b16R165
everything else is failing in oc10 as well

**Review by individual-it — APPROVED** [review]

**Pull request merged**